### PR TITLE
Add test coverage for `Mastodon::CLI::Accounts#prune`

### PR DESCRIPTION
**Pull request by danielmbrasil**
This PR adds tests for the `tootctl accounts prune` command.

### CODE REVIEW

**Comment by mjankowski** [comment]
@danielmbrasil - just want to say these various CLI coverage PRs are all great and very welcome!

Unrelated to this specific PR, how do feel about a pattern like this? (WIP branch from me to show idea) -- https://github.com/mastodon/mastodon/compare/main...mjankowski:mastodon:cli-specs-stdout-silence

I'd like to find some way to:

- Silence the output to STDOUT while rspec is running, which only happens when the expectation block has the `output` / `to_stdout` called
- Only run the CLI command once even when we have multiple assertions on it, mostly for spec suite speed reasons

**Comment by danielmbrasil** [comment]
Hey @mjankowski, thank you for your comment!
> how do feel about a pattern like this?

It looks great! That should reduce a lot of duplication and the tests look even more readable than before.

> Silence the output to STDOUT while rspec is running

Yeah, I've been looking for a way to silence it too, and I've come across this option:
```rb
allow($stdout).to receive(:write)
```
It seems to work fine and doesn't seem to break anything (even when there's an `output` / `to_stdout` expectation). However, I'm not sure if it's a good solution. What do you think?

> Only run the CLI command once even when we have multiple assertions on it

I don't know how that could be done. I've been using both `cli.invoke(:method)` and `cli.method` (when I have to mock/stub something). But, I'm not sure if there's a performance difference between those two option.

**Comment by mjankowski** [comment]
> It looks great! That should reduce a lot of duplication and the tests look even more readable than before.

Cool ... let me expand my WIP branch a bit and I'll submit that as a PR to get more feedback and see if the direction still feels good beyond just that one spec file.


> Yeah, I've been looking for a way to silence it too, and I've come across this option:
> 
> ```ruby
> allow($stdout).to receive(:write)
> ```

I did some research too and came across that approach in a few places as well. I think aesthetically and conceptually I like the appearance of the output/to_stdout block approach better than this approach of mocking `$stdout`. Why don't I push that WIP branch a bit further and then we can use that PR to discuss the approaches further, instead of hijacking this one.

> > Only run the CLI command once even when we have multiple assertions on it
> 
> I don't know how that could be done. I've been using both `cli.invoke(:method)` and `cli.method` (when I have to mock/stub something). But, I'm not sure if there's a performance difference between those two option.

I bet they are roughly the same. The `invoke` approach seems like a truer reproduction of how thor is actually running ... but when you have no arguments to pass in the only thing `invoke` is doing is calling the method, so it's probably fine in those cases to just call directly. If we wanted full consistency we'd have to use `invoke` everywhere because we will sometimes need to pass in options.

I think that the approach in my WIP branch of chaining together the expectations does achieve the goal of only calling each command method once, though - as opposed to calling the block to run the command in every single `it` block. I'll attempt to do a lightweight benchmark of this if I can change enough to have it be meaningful.

**Comment by mjankowski** [comment]
Having pushed the WIP branch a bit more locally ... it's actually sort of awkward to use that approach everywhere. Once they get even a little more complicated than what the cli/settings spec had it becomes less readable when chained together like that, even with well named methods.

Opening as alternate approach (stub stdout write) - https://github.com/mastodon/mastodon/pull/25211

File: spec/lib/mastodon/cli/accounts_spec.rb [hunk 1]
```diff
@@ -1361,4 +1361,83 @@
...
+    end
+
+    before do
+      Fabricate(:mention, account: mentioned_account, status: Fabricate(:status, account: Fabricate(:account)))
+      allow(cli).to receive(:parallelize_with_progress).and_yield(prunable_accounts[0])
+                                                       .and_yield(prunable_accounts[1])
+                                                       .and_yield(prunable_accounts[2])
+                                                       .and_yield(bot_account)
+                                                       .and_yield(group_account)
+                                                       .and_return([nil, 3])
```
[the remark this answers is not in the document]
Comment (ClearlyClaire) [reply spec/lib/mastodon/cli/accounts_spec.rb:1381]: See #27399

[the remark this answers is not in the document]
Comment (danielmbrasil) [reply spec/lib/mastodon/cli/accounts_spec.rb:1381]: > See #27399

Once it gets merged I'll rebase and update this PR.

[the remark this answers is not in the document]
Comment (ClearlyClaire) [reply spec/lib/mastodon/cli/accounts_spec.rb:1381]: #27399 has now been merged!

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**